internal/build/tar.go

```diff
@@ -2,28 +2,46 @@ package build
 
 import (
 	"archive/tar"
+	"bytes"
 	"context"
 	"fmt"
 	"io"
+	"log"
 	"os"
 	"path/filepath"
 	"strings"
 
 	opentracing "github.com/opentracing/opentracing-go"
 )
 
-func archiveDf(ctx context.Context, tw *tar.Writer, df Dockerfile) error {
+type archiveBuilder struct {
+	tw  *tar.Writer
+	buf *bytes.Buffer
+}
+
+func newArchiveBuilder() *archiveBuilder {
+	buf := new(bytes.Buffer)
+	tw := tar.NewWriter(buf)
+
+	return &archiveBuilder{tw: tw, buf: buf}
+}
+
+func (a *archiveBuilder) close() error {
+	return a.tw.Close()
+}
+
+func (a *archiveBuilder) archiveDf(ctx context.Context, df Dockerfile) error {
 	span, ctx := opentracing.StartSpanFromContext(ctx, "daemon-archiveDf")
 	defer span.Finish()
 	tarHeader := &tar.Header{
 		Name: "Dockerfile",
 		Size: int64(len(df)),
```

Comment:
i bet putting a recent `ModTime` in here will fix that problem with the unrealistic time stamp or whatever